transaction/src/plan/memo.rs

```diff
@@ -9,20 +9,30 @@ use rand::{CryptoRng, RngCore};
 
 #[derive(Clone, Debug)]
 pub struct MemoPlan {
-    pub plaintext: MemoPlaintext,
+    pub plaintext: [u8; MEMO_LEN_BYTES],
     pub key: PayloadKey,
 }
 
 impl MemoPlan {
     /// Create a new [`MemoPlan`].
-    pub fn new<R: CryptoRng + RngCore>(rng: &mut R, plaintext: MemoPlaintext) -> MemoPlan {
+    pub fn new<R: CryptoRng + RngCore>(
+        rng: &mut R,
+        plaintext_str: String,
+    ) -> Result<MemoPlan, anyhow::Error> {
+        if plaintext_str.as_bytes().len() > MEMO_LEN_BYTES {
+            return Err(anyhow::anyhow!(
+                "provided plaintext for memoplan exceeds maximum memo size"
+            ));
+        }
         let key = PayloadKey::random_key(rng);
-        MemoPlan { plaintext, key }
+        let mut plaintext = [0u8; MEMO_LEN_BYTES];
+        plaintext[..plaintext_str.as_bytes().len()].copy_from_slice(plaintext_str.as_bytes());
+        Ok(MemoPlan { plaintext, key })
     }
 
     /// Create a [`MemoCiphertext`] from the [`MemoPlan`].
-    pub fn memo(&self) -> MemoCiphertext {
-        self.plaintext.encrypt(self.key.clone())
+    pub fn memo(&self) -> Result<MemoCiphertext, anyhow::Error> {
+        MemoCiphertext::encrypt(&String::from_utf8_lossy(&self.plaintext), self.key.clone())
```

Comment:
As above, if we make the `MemoPlan` contain a `String`, then we don't need to do a conversion from UTF-8 here, which would be internally converted right back again in `MemoCiphertext::encrypt`.